source/dub/internal/utils.d

```diff
@@ -325,6 +325,92 @@ ubyte[] download(URL url, uint timeout = 8)
 	return download(url.toString(), timeout);
 }
 
+/**
+	Downloads a file from the specified URL with retry logic.
+
+	Downloads a file from the specified URL with up to 3 tries on failure
+	Throws: `Exception` if the download failed or `HTTPStatusException` on "unrecoverable failures" such as 404 not found.
+	Otherwise might throw anything else that `download` throws.
+	See_Also: download
+**/
+void retryDownload(URL url, NativePath filename)
+{
+	foreach(i; 0..3) {
+		version(DubUseCurl) {
+			try {
+				download(url, filename);
+				return;
+			}
+			catch(HTTPStatusException e) {
+				if (e.status == 404) throw e;
+				else {
+					logDebug("Failed to download %s (Attempt %s of 3)", url, i + 1);
+					if (i == 2) throw e;
+					else continue;
+				}
+			}
+			catch(CurlException e) {
```

Comment:
Why not putting this only in a `version` block?

Reply:
I am not sure whether I understood your question correctly.
Did you asked why I didn't put the the catch(CurlException) only
In a version scope?
I tried this but got syntax errors. It seems it isn't allowed.